controllers/capa_controller.go

```diff
@@ -154,27 +169,15 @@ func (r *CAPAClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{}, microerror.Mask(err)
 		}
 
-		finalizers := cluster.GetFinalizers()
-		if key.ContainsFinalizer(finalizers, key.FinalizerName) {
-			return ctrl.Result{
-				Requeue:      true,
-				RequeueAfter: time.Minute * 5,
-			}, nil
+		if created {
+			r.sendEvent(cluster, v1.EventTypeNormal, "IRSA", "IRSA bootstrap created")
 		}
 
-		controllerutil.AddFinalizer(cluster, key.FinalizerName)
-		err = r.Update(ctx, cluster)
-		if err != nil {
-			logger.Error(err, "failed to add finalizer on AWSCluster CR")
-			return ctrl.Result{}, microerror.Mask(err)
-		}
-		r.sendEvent(cluster, v1.EventTypeNormal, "IRSA", "IRSA bootstrap created")
+		return ctrl.Result{
```

Comment:
Isn't this the happy path? Why requeing after 5 minutes? Normally we don't want to requeue if things worked as expected.

Reply:
Left it as-is. There is unfortunately no comment, but I think automatic rescheduling is okay because the operator automatically updates allowed OIDC certificate thumbprints (see `EnsureOIDCProviders` function). If we change this, the controller may not be called at all anymore (in theory).

Reply:
The assumption from Andreas is correct 😃, we need to check constantly the OIDCProvider changes. Especially thumbprints have to be updated or if customer is adding new tags and wants to have those on the AWS resources we create for them.

Reply:
Thx! I'm adding a comment about that, and will then merge.